app/src/main/java/com/quran/labs/androidquran/ui/fragment/SuraListFragment.java

```diff
@@ -88,7 +92,11 @@ public void onPause() {
   public void onResume() {
     final Activity activity = getActivity();
     if (activity instanceof QuranActivity) {
-      QuranSettings settings = QuranSettings.getInstance(activity);
+      boolean newValueOfShowSuraTranslatedName = quranSettings.isShowSuraTranslatedName();
+      if (this.showSuraTranslatedName != newValueOfShowSuraTranslatedName) {
+        showHideSuraTranslatedName(newValueOfShowSuraTranslatedName);
+        this.showSuraTranslatedName = newValueOfShowSuraTranslatedName;
```

Comment:
what i meant is replace these 2 lines with something like:

```java
final QuranRow[] updatedElements = getSuraList();
adapter.setElements(updatedElements);
```

and in `onCreate`, make the adapter in a variable so you can call this setter on it. expose the setter on the adapter also. then you can just remove the show/add methods that you added below.

Reply:
Hi @ahmedre 
I got what do you want to say, but I believe that we don't have to repeat the construction of the SuraList by Calling getSuraList() again, instead of that, we can just modify the text of each Sura in the list by appending or removing the translated part.

I believe this implementation is more optimal than calling the getSuraList again and again.

Reply:
i agree that it's more optimal your way, but considering:

1. we can avoid adding extra code - so tradeoff for code readability
2. the extra price is only called when the person goes into the menu and chooses the option to show translation names - the number of people doing this is very small and will likely only do it once or twice at most.
3. the extra code means that in the future, if we change the way we're adding the translations (i.e. today, i guess we're manually making the string, which is not ideal and honestly we should fix this), then we'd have to modify this code - it also means other languages can't choose to something different - ex one language can use () one can use <>, one can do nothing at all.

so for those reasons, was suggesting to do it this way despite being slightly less optimal.

Reply:
got it, thanks for the clarification.